app/scenes/DocumentPermanentDelete.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+import { observer } from "mobx-react";
+import * as React from "react";
+import { useTranslation, Trans } from "react-i18next";
+import { useHistory } from "react-router-dom";
+import Document from "models/Document.js";
+import Button from "components/Button";
+import Flex from "components/Flex";
+import HelpText from "components/HelpText";
+import useStores from "hooks/useStores";
+import { collectionUrl } from "utils/routeHelpers";
+
+type Props = {
+  document: Document,
+  onSubmit: () => void,
+};
```

Comment:
```suggestion
type Props = {|
  document: Document,
  onSubmit: () => void,
|};
```

Sealing the object stops random props from being passed in without a warning

Reply:
learned something new!